barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/mock_transcript.hpp

```diff
@@ -0,0 +1,100 @@
+#include "barretenberg/ecc/curves/grumpkin/grumpkin.hpp"
+#include "barretenberg/numeric/uint256/uint256.hpp"
+#include <vector>
+namespace bb {
+/**
+ * @brief Mock transcript class used by IPA tests and fuzzer
+ *
+ * @details This transcript can send previously determined challenges instead of ones generated by Fiat Shamir. It can
+ * also store elements received from the prover
+ *
+ */
+class MockTranscript {
+  public:
+    // Vector of challenges sent to the verifier
+    std::vector<uint256_t> challenges;
+    // Vector of group elements received from the prover / sent to the verifier
+    std::vector<bb::curve::Grumpkin::AffineElement> group_elements;
+    // Vector of field elements received from the prover / sent to the verifier. uint256_t is used to ignore field type
+    std::vector<uint256_t> field_elements;
```

Comment:
won't conversions get confused if you ignore field types? why not templating if you want different fields?

Reply:
wdym by "confused"? Both base and scalar field elements can be easily converted